sources/core/objects/Floor.cpp

```diff
@@ -28,7 +28,6 @@ void Floor::AfterWorldCreation()
 {
     ITurf::AfterWorldCreation();
     SetOpen(open_);
-    SetState(floor_type_);
```

Comment:
Well, now it is impossible to set the floor type in the map editor.
You still should override the `AfterWorldCreation` function in the `Plating`, it is the best way I think.
It should be something like
```
void Plating::AfterWorldCreation()
{
    Floor::AfterWorldCreation();
    // Hack
    SetState("plating");
    floor_type_ = "floor";
}
```